### PR TITLE
Lab7 Completed with Challenge.

### PR DESCRIPTION
**Pull request by Jacky1683**

### CODE REVIEW

File: src/main/java/lab7/CsvDataService.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package lab7;
+
+import com.opencsv.bean.CsvToBeanBuilder;
+import com.opencsv.exceptions.CsvException;
+
+import java.io.FileReader;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class CsvDataService implements DataService {
```
Comment (rmartin-sc) [inline src/main/java/lab7/CsvDataService.java:10]: Nice!

File: src/test/java/lab7/MainTest.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+    @Test
+    void testSearchRecipesReturnsEmptyWhenNoMatch() {
+        DataService mock = () -> List.of(
+                new Recipe(1, "Tofu Stir Fry", "Vegan meal", "", 4, 10, 10, 20)
+        );
+        var results = Main.searchRecipes("chicken", mock);
+        assertTrue(results.isEmpty());
+    }
+
```
Comment (rmartin-sc) [inline src/test/java/lab7/MainTest.java:93]: You should include a test that exceptions thrown by the DataService are handled gracefully by returning an empty list
